src/watchmaker/workers/salt.py

```diff
@@ -980,12 +1005,17 @@ def install(self):
         salt_running = False
         salt_enabled = False
         salt_svc = 'salt-minion'
+        self.salt_call = SaltWindows._get_salt_call()
         if os.path.exists(self.salt_call):
             salt_running, salt_enabled = self.service_status(salt_svc)
-        self._install_package()
-        if self.pip_install:
-            self._install_pip_packages()
-        self.salt_call = SaltWindows._get_salt_call()
+            self.log.info('Skipping installation, already installed.')
+        else:
+            self.log.info('Installing Salt minion...')
+            self._install_package()
+            if self.pip_install:
+                self._install_pip_packages()
+            self.salt_call = SaltWindows._get_salt_call()
+            self.log.info("Salt minion installation completed successfully.")
```

Comment:
I don't see why this change is needed either. Can you explain or revert?

```suggestion
        self._install_package()
        if self.pip_install:
            self._install_pip_packages()
        self.salt_call = SaltWindows._get_salt_call()
```

Reply:
I believed this was the initial requirement but I can revert it if its not needed.
This block of code checks if the Salt minion is already installed. If it is already installed, it logs a message saying "Skipping=installation, already installed." and sets the salt_running and salt_enabled variables to True or False depending on whether the Salt service is running and enabled. If the Salt minion is not already installed, it logs a message saying "Installing Salt minion...", installs the package by calling the _install_package() method, installs any pip packages specified by calling _install_pip_packages() if the pip_install attribute is set to True, and sets the salt_call attribute to the path of the salt-call executable. Finally, it logs a message saying "Salt minion installation completed successfully.”